### ISSUE TITLE
Performance: Fabricator FPS drops with linked containers.

### ISSUE THREAD

**Issue by Ek-Videogames** [Regalis11/Barotrauma#9284]
- Build 0.17.15.0 Windows
- [X] I have searched the issue tracker to check if the issue has already been reported.
*I mean - the other ticket is closed, and this 'could' be caused by a new bug if the old one is not to blame.
- [ ] My issue happened while using mods.

###Description
Issue #6735 
Fabricator GUI draw problems seem to be back again.

### Reproduction steps
Not including "Happened while using mods" because this can also be reproduced by linking a bunch of vanilla lockers and storage-containers.

However possible reproduction with modlist: (and sub with linked containers)
> EK | Armory
> EK | Dockyard
> EK | Gunnery
> EK | Utility
> BaroDrama V3.1
> Sounds Of Europa
> U.N.T Aggressor

### Additional Information
Possibly the increasingly common convention of having items with multiple alternate crafting recipes
(ammo boxes, welding tanks, etc)
is straining a different part of fabricator GUI's logic?

**Comment by Regalis11** [comment]
I'm afraid I could not reproduce this. Tested by filling the cabinet in U.N.T Aggressor with storage containers: the FPS stays solid at 60. 
![image](https://user-images.githubusercontent.com/14351070/175976004-496ec889-68fe-421e-b311-61b7998286d1.png)

**Comment by Regalis11** [comment]
No known steps to repro, closing

**Comment by NilanthAnimosus** [comment]
well, I tried modifying a winterhalter submarine in 0.8.15.0, Filled them up with items related to crafting and some not that are usually just deconstructed. Probably at least exceeding well over 5000 items added in just two large linked containers filled to the brim with storage containers.

I tried with and without mods (ensuring they had as many overwrites as possible to vanilla content and some items being overwritten more than once on top of new content), and played with fabricators filtering on items.

My specific mods of choice were:
- Sounds of Europa
- Better Vanilla Weapons
- Submarine Weapon Sound Makeover
- EK | Armory
- EK | Utility
- EK | Gunnery
- EK | Dockyard
- Submarine


just to note on my hardware since I feel its relevant:
haswell i7-5930K 6 core (12 thread/logical core) processor running at 4.1Ghz
memory (DDR3, 32gb's at 2400Mhz)
2080 Ti (11GB graphics memory)
game running off SSD (3gb-3.5gb/s read/write limited by PCI Gen)

It sounds sort of similar to the original issue but I did have to go to an excessive amount (two linked large cabinets fully filled in every possible slot), but I had massive spikes when scrolling down the huge item list like it was loading stuff into memory (going as low as 2FPS while scrolling at times) from the modded check - and I figure it is maybe loading graphics on the fly for the items in the list.

my framerate didn't tank or anything like that permanently (Though if it did, it was 10 or less frames change from . and searching was fine and had a barely measurable effect compared to first time list scrolling. makes me wonder if that issue is more tied to hardware as well.

Hosted via "Ingame hosting" on the submarine with sandbox gamemode but also using my debug copy (not release):
Waited a bit until FPS felt stable, lighting/los on. around a 42 - 48 FPS
Open deconstructor for both containers without fab UI (only decon UI with it) - around 35 - 37FPS
open fabricator for both containers with fab UI (instead of decon) - around 29 - 32 FPS.
Scroll down fabricator list very quickly (via mousewheel) gave me a few game freezes (1-2fps, 6fps, 11 fps etc) and my average was capable of dropping to sub 20 FPS.

Given the sheer amount of items it required to get my framerate to drop this heavilly (the sub had like 7000 total items when I was done from some 1000-2000 items), and only reached very low levels when rapidly scrolling the entire list, I'm not sure if this should be reopened and looked into given I was dropping managable frames, but then perhaps that is more than 80% of someone else's frames whom already does not run at the best framerate and this drop from the high item quantity does not strictly require mods, though mods definitely effects scrolling the fabricator list.

Below is the winterhalter submarine I edited with a ModList but its more just sheer quantity of items draining performance somewhat when having linked containers to a fabricator, and a drain in performance when scrolling a large list of modded/vanilla items in the fabricator UI at first viewing.


[Winterhalter - fab UI item.zip](https://github.com/Regalis11/Barotrauma/files/9138846/Winterhalter.-.fab.UI.item.zip)
